src/python/turicreate/toolkits/_tf_utils.py

```diff
@@ -8,6 +8,17 @@
 from __future__ import division as _
 from __future__ import absolute_import as _
 import numpy as np
+
+
+def get_gpu_names():
+	"""
+	Gets the available GPU names.
+	"""
+	import turicreate.toolkits.libtctensorflow
+	from tensorflow.python.client import device_lib
+	local_device_protos = device_lib.list_local_devices()
+	gpu_names = [x.name for x in local_device_protos if x.device_type == 'GPU']
+	return gpu_names if not None else []
```

Comment:
Is it possible for `gpu_names` to ever be `None`? I think even if `local_device_protos` is empty, `gpu_names` will be `[]`.

Reply:
You are right because it will at least be CPU which would make `gpu_names` at least an empty list.

Reply:
Then should we just return `gpu_names`?

Besides, doesn't `gpu_names if not None else []` always return `gpu_names`? Because `not None` is always `True`?

Reply:
I changed this to be compatible with TF 2.0 API. I checked it returns an empty list if no device is available so removed this condition.